Library/Homebrew/formulary.rb

```diff
@@ -105,7 +105,17 @@ def load_file
   # Loads formulae from bottles.
   class BottleLoader < FormulaLoader
     def initialize(bottle_name)
-      @bottle_filename = Pathname(bottle_name).realpath
+      case bottle_name
+      when %r{(https?|ftp|file)://}
+        resource = Resource.new bottle_name[%r{([^/]+)-}, 1] { url bottle_name }
```

Comment:
Done. The name of the formula is found between the last slash and the last hyphen.